integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/car/deployment/test/CAppDeploymentOrderTest.java

```diff
@@ -0,0 +1,180 @@
+*under the License.
+*/
+package org.wso2.carbon.esb.car.deployment.test;
+
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.ArrayUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.extensions.servers.ftpserver.FTPServerManager;
+import org.wso2.carbon.integration.common.admin.client.ApplicationAdminClient;
+import org.wso2.carbon.integration.common.admin.client.CarbonAppUploaderClient;
+import org.wso2.carbon.integration.common.admin.client.LogViewerClient;
+import org.wso2.carbon.logging.view.stub.LogViewerLogViewerException;
+import org.wso2.carbon.logging.view.stub.types.carbon.LogEvent;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.Utils;
+
+import javax.activation.DataHandler;
+import java.io.File;
+import java.net.URL;
+import java.rmi.RemoteException;
+import java.util.Calendar;
+import java.util.Stack;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class tests whether the artifacts in deployed car file is in order
+ * Related issue: https://github.com/wso2/product-ei/issues/1261
+ */
+public class CAppDeploymentOrderTest extends ESBIntegrationTest {
+    private CarbonAppUploaderClient carbonAppUploaderClient;
+    private ApplicationAdminClient applicationAdminClient;
+    private final int MAX_TIME = 120000;
+    private final String carFileName = "esb-deployment-car_1.0.0";
+    private FTPServerManager ftpServerManager;
+    private String FTPUsername;
+    private String FTPPassword;
+    private File FTPFolder;
+    private File inputFolder;
+    private LogViewerClient logViewerClient;
+    private String pathToFtpDir;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        runFTPServerForInboundTest();
+        super.init();
+        logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+    }
+
+    @Test(groups = "wso2.esb", enabled = true, description = "Test whether proxy service get deployed through capp in order")
+    protected void carFileDeploymentOrderTest() throws Exception {
+        logViewerClient.clearLogs();
+        carbonAppUploaderClient = new CarbonAppUploaderClient(context.getContextUrls().getBackEndUrl(), getSessionCookie());
+        carbonAppUploaderClient.uploadCarbonAppArtifact("esb-deployment-car_1.0.0.car"
+                , new DataHandler(new URL("file:" + File.separator + File.separator + getESBResourceLocation()
+                        + File.separator + "car" + File.separator + "esb-deployment-car_1.0.0.car")));
+        applicationAdminClient = new ApplicationAdminClient(context.getContextUrls().getBackEndUrl(), getSessionCookie());
+        Assert.assertTrue(isCarFileDeployed(carFileName), "Car file deployment failed");
+        TimeUnit.SECONDS.sleep(5);
```

Comment:
Do we need this sleep since isCarFileDeployed handle the deployment ?

Reply:
Fixed in fd4486d8cd20c8944413168817517ebb691f3594